test/unit/gateways/merchant_partners_test.rb

```diff
@@ -0,0 +1,792 @@
+      @gateway.void("5d53a33d960c46d00f5dc061947d998c")
+    end.respond_with(failed_void_response)
+
+    assert_failure response
+  end
+
+  def test_successful_refund
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card)
+    end.respond_with(successful_purchase_response)
+
+    assert_success response
+
+    refund = stub_comms do
+      @gateway.refund(@amount, response.authorization)
+    end.check_request do |endpoint, data, headers|
+      parse(data) do |doc|
+        assert_not_nil root = doc.at_xpath(@request_root)
+        assert_equal @gateway.options[:account_id], root.at_xpath("//acctid").content
+        assert_equal @gateway.options[:merchant_pin], root.at_xpath("//merchantpin").content
+        assert_equal response.authorization, root.at_xpath("//historykeyid").content
+        assert_equal "4", root.at_xpath("//service").content
+        assert_equal "1.00", root.at_xpath("//amount").content
+      end
+    end.respond_with(successful_refund_response)
+
+    assert_success refund
+  end
+
+  def test_failed_refund
+    response = stub_comms do
+      @gateway.refund(nil, "")
+    end.respond_with(failed_refund_response)
+
+    assert_failure response
+  end
+
+  def test_successful_credit
+    response = stub_comms do
+      @gateway.credit(@amount, @credit_card)
+    end.check_request do |endpoint, data, headers|
+      parse(data) do |doc|
+        assert_not_nil root = doc.at_xpath(@request_root)
+        assert_equal @gateway.options[:account_id], root.at_xpath("//acctid").content
+        assert_equal @gateway.options[:merchant_pin], root.at_xpath("//merchantpin").content
+        assert_equal @credit_card.name, root.at_xpath("//ccname").content
+        assert_equal @credit_card.number, root.at_xpath("//ccnum").content
+        assert_equal @credit_card.verification_value, root.at_xpath("//cvv2").content
+        assert_equal "6", root.at_xpath("//service").content
+        assert_equal "1.00", root.at_xpath("//amount").content
+      end
+    end.respond_with(successful_credit_response)
+
+    assert_success response
+    assert response.test?
+  end
+
+  def test_failed_credit
+    response = stub_comms do
+      @gateway.credit(@amount, @credit_card)
+    end.respond_with(failed_credit_response)
+
+    assert_failure response
+    assert_equal "Invalid account number", response.message
+    assert response.error_code.start_with?("1102")
+    assert response.test?
+  end
+
+  def test_successful_verify
+    response = stub_comms do
+      @gateway.verify(@credit_card)
+    end.respond_with(successful_authorize_response, failed_void_response)
+    assert_success response
+    assert_equal "Succeeded", response.message
+  end
+
+  def test_failed_verify
+    response = stub_comms do
+      @gateway.verify(@credit_card)
+    end.respond_with(failed_authorize_response, successful_void_response)
+    assert_failure response
+    assert_equal "Invalid account number", response.message
+  end
+
```

Comment:
I'm still thinking it's worth it to have a `test_successful_verify_with_failed_void` unit test to ensure we succeed even if the void fails.  Whatcha think?